test/lib/jdk/test/lib/security/NSSArtifactFetcher.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.test.lib.security;
+
+import jdk.test.lib.Platform;
+import jdk.test.lib.security.artifacts.ThirdPartyArtifacts;
+import jtreg.SkippedException;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Properties;
+import java.util.stream.Stream;
+
+public class NSSArtifactFetcher {
+
+    public static final String DEFAULT_NSS_LIBRARY =  "softokn3";
+
+     private static Class<?> getNssLibClass(String osId){
+        switch (osId) {
+            case "Windows-amd64-64":
+                return ThirdPartyArtifacts.WINDOWS_X64.class;
+
+            case "MacOSX-x86_64-64":
+                return ThirdPartyArtifacts.MACOSX_X64.class;
+
+            case "MacOSX-aarch64-64":
+                return ThirdPartyArtifacts.MACOSX_AARCH64.class;
+
+            case "Linux-amd64-64":
+                if (Platform.isOracleLinux7()) {
+                    throw new SkippedException("Skipping Oracle Linux prior to v8");
+                } else {
+                    return ThirdPartyArtifacts.LINUX_X64.class;
+                }
+
+            case "Linux-aarch64-64":
+                if (Platform.isOracleLinux7()) {
+                    throw new SkippedException("Skipping Oracle Linux prior to v8");
+                } else {
+                    return ThirdPartyArtifacts.LINUX_AARCH64.class;
+                }
+            default:
+                return null;
+        }
+    }
+
+    public static String getNSSLibDir(String library) throws Exception {
```

Comment:
You have a `getNSSLibDir()` method and one called `getNSSDirectory()`.  Please add Javadocs to explain what's what.